python_modules/dagster/dagster/core/test_utils.py

```diff
@@ -562,3 +563,30 @@ async def run():
     counts = counter.counts()
     assert counts["foo"] == 20
     assert counts["bar"] == 10
+
+
+class MockThreadPoolExecutor:
+    def __init__(self, **kwargs):
+        pass
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        pass
+
+    def submit(self, fn, *args, **kwargs):
+        # execute functions in series without creating threads
+        # for easier unit testing
```

Comment:
nit: would prefer this to be as a top-level docstring on the class, since this basically summarizes the class